sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -323,11 +374,20 @@ private[joins] object UnsafeHashedRelation {
     // Create a mapping of buildKeys -> rows
     val keyGenerator = UnsafeProjection.create(key)
     var numFields = 0
+    val numKeys = key.length
+    val isOriginalInputEmpty = !input.hasNext
+    var allNullColumnKeyExistsInOriginalInput: Boolean = false
     while (input.hasNext) {
       val row = input.next().asInstanceOf[UnsafeRow]
       numFields = row.numFields()
       val key = keyGenerator(row)
-      if (!key.anyNull) {
+      if (isNullAware &&
+        !allNullColumnKeyExistsInOriginalInput &&
+        (0 until numKeys).forall(key.isNullAt)) {
```

Comment:
Dumb question, which I know I have asked before: Can `numKeys` be more than 1 now given that we are only considering the single key case ? If not, should this `allNullColumnKeyExistsInOriginalInput` be changed to `keyWithSingleNullColumnExistsInOriginalInput` (as an example name) in addition to changing the `forall` above to simply check for `key.isNullAll(0)` ? 

If you do consider making this change, just a reminder to change `allNullColumnKeyExistsInOriginalInput` in the `LongHashedRelation` too

Reply:
I think i will still insist using the naming allNullColumnKeyExistsInOriginalInput,  but i will change the 
(0 until numKeys).forall(key.isNullAt) into key.isNullAt(0)

Reply:
done